src/mlpack/methods/adaboost/adaboost_classify_main.cpp

```diff
@@ -31,16 +31,15 @@
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
-#include <mlpack/prereqs.hpp>
-#include <mlpack/core/util/io.hpp>
+#include <mlpack/core.hpp>
 
 #ifdef BINDING_NAME
   #undef BINDING_NAME
 #endif
```

Comment:
@rcurtin the `#ifdef .. #endif` block can be replaced by simply `#undef BINDING_NAME`.

Extract from https://en.cppreference.com/w/c/preprocessor/replace  
`The #undef directive undefines the identifier, that is it cancels the previous definition of the identifier by #define directive. If the identifier does not have an associated macro, the directive is ignored.`

In other words, `#undef BLAH` works even if `BLAH` is not defined beforehand.

(There are further instances of the `#ifdef .. #endif` block in other files. Commenting only on this instance for brevity)

Reply:
Thanks, I handled this with an awk script in c4e05d4.  It would have been too tedious to do by hand :smile: